cutil/logger/logger.go

```diff
@@ -137,6 +125,20 @@ func Warning(format string, a ...interface{}) {
 	}
 }
 
+func extractLoggerArgs(format string, a ...interface{}) ([]interface{}, io.Writer) {
```

Comment:
I was trying to find out if there is any documentation for logging, but couldn't find. Just trying to wrap my head around this, is the expectation is when someone logs using this package, they can pass in an io.writer as the last argument?  Ex -

logger.Log("Created something %s", id, myLogger) 

Looks like we are expecting an io.writer in the end of arguments, which makes it really confusing to use. If we are expecting a writer in each log call, we can accept it as a first argument same as what fmt.Fprintf does in std lib or it can be nil. 
But, I don't thikn passing a writer in each call is a good option. This is something I would pass in for the whole package during init and expect all my logs to go there.